packages/svelte2tsx/src/htmlxtojsx_v2/nodes/Element.ts

```diff
@@ -244,6 +238,96 @@ export class Element {
         }
     }
 
+    private getStartTransformation(): TransformationArray {
+        const createElement = `${this.typingsNamespace}.createElement`;
+        const addActions = () => {
+            if (this.actionIdentifiers.length) {
+                return `, __sveltets_2_union(${this.actionIdentifiers.join(',')})`;
+            } else {
+                return '';
+            }
+        };
+
+        switch (this.node.name) {
```

Comment:
From what I can see the `if-else` statements are quite repetitive and error-prone on changes.
I would write it like this:

```ts
const getCreateElementStatement = (name: string): string => {
   // switch-case statement with just the else branches
   // as a single string and not as an array (from what I see it is always 1 element)
   // and without the leading '{'
}

const createElementStatement = getCreateElementStatement(this.node.name)
const variable = this.referencedName ? `const ${this._name} = ` : ''

return [`{ ${variable}${createElementStatement}`]
```

Reply:
Yeah I was thinking about cleaning this up, too, this is a good outline of the solution